jaeger-core/src/main/java/com/uber/jaeger/Span.java

```diff
@@ -267,19 +267,52 @@ private Span setTagAsObject(String key, Object value) {
   }
 
   @Override
+  public Span log(Map<String, ?> fields) {
+    return log(tracer.clock().currentTimeMicros(), fields);
+  }
+
+  @Override
+  public Span log(long timestampMicroseconds, Map<String, ?> fields) {
+    synchronized (this) {
+      if (fields == null) {
+        return this;
+      }
+      if (context.isSampled()) {
+        if (logs == null) {
+          this.logs = new ArrayList<>();
+        }
+        logs.add(new LogData(timestampMicroseconds, fields));
+      }
+      return this;
+    }
+  }
+
+  @Override
+  public Span log(String event) {
+    return log(tracer.clock().currentTimeMicros(), event, null);
+  }
+
+  @Override
+  public Span log(long timestampMicroseconds, String event) {
+    return log(timestampMicroseconds, event, null);
+  }
+
+  @Override
   public Span log(String message, /* @Nullable */ Object payload) {
     return log(tracer.clock().currentTimeMicros(), message, payload);
   }
 
   @Override
-  public Span log(long instantMicroseconds, String message, /* @Nullable */ Object payload) {
+  public Span log(long timestampMicroseconds, String message, /* @Nullable */ Object payload) {
```

Comment:
I wasn't suggesting it. Payload as map should be stringified as any other object.